### PR TITLE
fix contorl contract bug(to v0.19.17)

### PR DESCRIPTION
**Pull request by BananaLF**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/keeper/proposal.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func (k Keeper) CheckMsgSubmitProposal(ctx sdk.Context, msg govTypes.MsgSubmitPr
 				if bc == nil {
 					return types.ErrBlockedContractMethodIsNotExist(content.ContractList[i].Address,types.ErrorContractMethodBlockedIsNotExist)
 				}
-				if err := bc.BlockMethods.DeleteContractMethodMap(content.ContractList[i].BlockMethods);err != nil {
+				if _,err := bc.BlockMethods.DeleteContractMethodMap(content.ContractList[i].BlockMethods);err != nil {
```
Comment (xiangjianmeng) [inline x/evm/keeper/proposal.go:62]: format

File: x/evm/types/address_list.go [hunk 2]
```diff
@@ -112,6 +118,14 @@ func (bc BlockedContract) String() string {
 //ContractMethods is the list of blocked contract method
 type ContractMethods []ContractMethod
 
+func SortContractMethods(cms []ContractMethod)  {
```
Comment (xiangjianmeng) [inline x/evm/types/address_list.go:121]: sortContractMethods

File: x/evm/types/address_list.go [hunk 3]
```diff
@@ -232,7 +249,7 @@ func (cmbc *ContractMethodBlockedCache) SetContractMethod(keyData []byte, bc Con
 	cmbc.cache.Add(key, bc)
 }
 
-func BlockedContractListIsEqual(src, dst BlockedContractList) bool {
+func BlockedContractListIsEqual(t *testing.T,src, dst BlockedContractList) bool {
```
Comment (xiangjianmeng) [inline x/evm/types/address_list.go:252]: test code should not appear here

File: x/evm/types/address_list.go [hunk 4]
```diff
@@ -77,7 +81,9 @@ type BlockedContract struct {
 
 // NewBlockContract return point of BlockedContract
 func NewBlockContract(addr sdk.AccAddress, methods ContractMethods) *BlockedContract {
-	return &BlockedContract{Address: addr, BlockMethods: methods}
+	bm := make([]ContractMethod,len(methods))
+	copy(bm,methods)
```
Comment (xiangjianmeng) [inline x/evm/types/address_list.go:85]: it is suggested to implement copy in function `GetContractMethod`.

**Review by zhongqiuwood — APPROVED** [review]

**Pull request merged**